Makefile

```
@@ -28,16 +28,15 @@ unittest:
 
 .PHONY: lint
 lint:
-	$(PYTHON) -m pylint --rcfile .pylintrc $(PYTHON_SOURCE_DIRS)
+	ruff check $(PYTHON_SOURCE_DIRS)
 
 .PHONY: mypy
 mypy:
 	$(PYTHON) -m mypy --show-error-codes $(PYTHON_SOURCE_DIRS)
 
 .PHONY: fmt
 fmt:
-	isort $(PYTHON_SOURCE_DIRS)
-	black $(PYTHON_SOURCE_DIRS)
+	ruff format $(PYTHON_SOURCE_DIRS)
```

```suggestion
        ruff check --fix $(PYTHON_SOURCE_DIRS)
	ruff format $(PYTHON_SOURCE_DIRS)
```

suggestion: Also allow auto-fixing.

---

Added, limited to isort rules, as I don't think the `fmt` target should fix other violations.

---

That's OK, but now I'm curious as to why 😄 In my experience the ruff auto-fixers exceedingly rarely does the wrong thing.

---

Yes, it is not about that, but rather about semantics. I think that a target that is supposed to format code should not make unrelated changes.